code/game/objects/items/food/pie.dm

```diff
@@ -0,0 +1,267 @@
+	if(isliving(loc)) //someone caught us!
+		return
+	var/turf/T = get_turf(hit_atom)
+	new/obj/effect/decal/cleanable/food/pie_smudge(T)
+	if(reagents?.total_volume)
+		reagents.expose(hit_atom, TOUCH)
+	if(isliving(hit_atom))
+		var/mob/living/L = hit_atom
+		if(stunning)
+			L.Paralyze(20) //splat!
+		L.adjust_blurriness(1)
+		L.visible_message("<span class='warning'>[L] is creamed by [src]!</span>", "<span class='userdanger'>You've been creamed by [src]!</span>")
+		playsound(L, "desecration", 50, TRUE)
+	if(is_type_in_typecache(hit_atom, GLOB.creamable))
+		hit_atom.AddComponent(/datum/component/creamed, src)
+	qdel(src)
+
+/obj/item/food/pie/cream/nostun
+	stunning = FALSE
+
+/obj/item/food/pie/berryclafoutis
+	name = "berry clafoutis"
+	desc = "No black birds, this is a good sign."
+	icon_state = "berryclafoutis"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 11, /datum/reagent/consumable/berryjuice = 5, /datum/reagent/consumable/nutriment/vitamin = 4)
+	tastes = list("pie" = 1, "blackberries" = 1)
+	foodtypes = GRAIN | FRUIT | SUGAR
+
+/obj/item/food/pie/bearypie
+	name = "beary pie"
+	desc = "No brown bears, this is a good sign."
+	icon_state = "bearypie"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 12, /datum/reagent/consumable/nutriment/protein = 5, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("pie" = 1, "meat" = 1, "salmon" = 1)
+	foodtypes = GRAIN | SUGAR | MEAT | FRUIT
+
+/obj/item/food/pie/meatpie
+	name = "meat-pie"
+	icon_state = "meatpie"
+	desc = "An old barber recipe, very delicious!"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 10, /datum/reagent/consumable/nutriment/vitamin = 4, /datum/reagent/consumable/nutriment/protein = 2)
+	tastes = list("pie" = 1, "meat" = 1)
+	foodtypes = GRAIN | MEAT
+
+/obj/item/food/pie/tofupie
+	name = "tofu-pie"
+	icon_state = "meatpie"
+	desc = "A delicious tofu pie."
+	food_reagents = list(/datum/reagent/consumable/nutriment = 11, /datum/reagent/consumable/nutriment/protein = 1, /datum/reagent/consumable/nutriment/vitamin = 6)
+	tastes = list("pie" = 1, "tofu" = 1)
+	foodtypes = GRAIN
+
+/obj/item/food/pie/amanita_pie
+	name = "amanita pie"
+	desc = "Sweet and tasty poison pie."
+	icon_state = "amanita_pie"
+	bite_consumption = 4
+	food_reagents = list(/datum/reagent/consumable/nutriment = 6, /datum/reagent/toxin/amatoxin = 3, /datum/reagent/drug/mushroomhallucinogen = 1, /datum/reagent/consumable/nutriment/vitamin = 4)
+	tastes = list("pie" = 1, "mushroom" = 1)
+	foodtypes = GRAIN | VEGETABLES | TOXIC | GROSS
+
+/obj/item/food/pie/plump_pie
+	name = "plump pie"
+	desc = "I bet you love stuff made out of plump helmets!"
+	icon_state = "plump_pie"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 11, /datum/reagent/consumable/nutriment/vitamin = 4)
+	tastes = list("pie" = 1, "mushroom" = 1)
+	foodtypes = GRAIN | VEGETABLES
+
+/obj/item/food/pie/plump_pie/Initialize()
+	. = ..()
+	var/fey = prob(10)
+	if(fey)
```

Comment:
Initialize needs to call parent last in this case. Food reagents are added to the component on init. Also what is up with the two if(fey" checks and omnizine additions